### ISSUE TITLE
Fastboot 1.1.3 - Cannot find module 'moment'

### ISSUE THREAD

**Issue by ehubbell** [jasonmit/ember-cli-moment-shim#165]
I’m running into a couple of dependency / module issues with this library when running the `fastboot-app-server` locally. When I have `includeTimezone: 'all'` listed in my environment file, build the ember app, and then start my node server using `PORT=8000 node server.js` I get the following error:

```
There was an error running your app in fastboot. More info about the error:
 Error: Cannot find module 'moment-timezone'
    at Function.Module._resolveFilename (module.js:555:15)
    at Function.Module._load (module.js:482:25)
    at Module.require (module.js:604:17)
    at require (internal/module.js:11:18)
    at Object.require (/Users/erichubbell/Sites/joyrides/fastboot-app-server/node_modules/fastboot/src/ember-app.js:130:18)
    at Module.callback (/Users/erichubbell/Sites/startup-playbooks/startup-playbooks-web/dist/moment/fastboot-moment-timezone.js:4:34)
```

When I remove `includeTimezone: 'all'` from my environment file, I get the following:

```
There was an error running your app in fastboot. More info about the error:
 Error: Cannot find module 'moment'
    at Function.Module._resolveFilename (module.js:555:15)
    at Function.Module._load (module.js:482:25)
    at Module.require (module.js:604:17)
    at require (internal/module.js:11:18)
    at Object.require (/Users/erichubbell/Sites/joyrides/fastboot-app-server/node_modules/fastboot/src/ember-app.js:130:18)
    at Module.callback (/Users/erichubbell/Sites/startup-playbooks/startup-playbooks-web/dist/moment/fastboot-moment.js:4:34)
```

I've tried several different version combinations of `ember-cli-fastboot`, `ember-moment`, and `ember-cli-moment-shim` in an attempt to fix this. I've also tried whitelisting those libraries and including them as dependencies in my `package.json` file. I also see the `dist/moment` directory with `fastboot-moment-timezone.js` and `fastboot-moment.js` files. 

If I remove `import moment from 'moment';` and any mentions of `moment()` from the component that's displayed on the homepage of my app - Fastboot loads fine. It's only when I add the import statement or call to `moment()` that Fastboot breaks. 

I've read over the ember-fastboot docs as they pertain to dependencies pretty thoroughly and am unsure what's causing this and / or how to troubleshoot further. Any thoughts on how to fix this Or, does anyone know a working combination of `ember-source`, `ember-cli-fastboot`, and `ember-cli-moment-shim`? Here's my current `package.json` file for reference and thanks in advance!

```
{
  "name": "startup-playbooks-web",
  "version": "0.0.0",
  "private": true,
  "description": "Small description for startup-playbooks-web goes here",
  "license": "MIT",
  "author": "",
  "directories": {
    "doc": "doc",
    "test": "tests"
  },
  "repository": "",
  "scripts": {
    "build": "ember build",
    "lint:js": "eslint ./*.js app config lib server tests",
    "start": "ember serve",
    "test": "ember test"
  },
  "devDependencies": {
    "bootstrap": "^4.0.0",
    "broccoli-asset-rev": "^2.4.5",
    "ember-ajax": "^3.0.0",
    "ember-cli": "~3.1.2",
    "ember-cli-app-version": "^3.0.0",
    "ember-cli-babel": "^6.6.0",
    "ember-cli-bootstrap-4": "^0.5.6",
    "ember-cli-dependency-checker": "^2.0.0",
    "ember-cli-eslint": "^4.2.1",
    "ember-cli-fastboot": "1.1.3",
    "ember-cli-flash": "^1.6.3",
    "ember-cli-google-fonts": "^2.16.2",
    "ember-cli-head": "^0.4.0",
    "ember-cli-htmlbars": "^2.0.1",
    "ember-cli-htmlbars-inline-precompile": "^1.0.0",
    "ember-cli-inject-live-reload": "^1.4.1",
    "ember-cli-moment-shim": "^3.5.0",
    "ember-cli-qunit": "^4.1.1",
    "ember-cli-sass": "^7.1.7",
    "ember-cli-shims": "^1.2.0",
    "ember-cli-sri": "^2.1.0",
    "ember-cli-uglify": "^2.0.0",
    "ember-cli-update": "^0.21.4",
    "ember-data": "~3.1.0",
    "ember-export-application-global": "^2.0.0",
    "ember-fetch": "^3.4.5",
    "ember-font-awesome": "^4.0.0-rc.2",
    "ember-load-initializers": "^1.0.0",
    "ember-maybe-import-regenerator": "^0.1.6",
    "ember-moment": "7.5.0",
    "ember-resolver": "^4.0.0",
    "ember-side-menu": "0.0.15",
    "ember-simple-auth": "^1.5.1",
    "ember-source": "~3.1.0",
    "ember-welcome-page": "^3.0.0",
    "eslint-plugin-ember": "^5.0.0",
    "loader.js": "^4.2.3",
    "popper.js": "^1.12.9"
  },
  "engines": {
    "node": "^4.5 || 6.* || >= 7.*"
  },
  "dependencies": {},
  "fastbootDependencies": [
    "fs",
    "path"
  ]
}
```

**Comment by kiwiupover** [comment]
@ehubbell 

you need to add `moment` to your `fastbootDependencies`

```json
"fastbootDependencies": [
    "moment",
    "fs",
    "path"
  ]
```

**Comment by sdhull** [comment]
@kiwiupover I get this same error. I tried adding `moment` to `fastbootDependencies` and it still doesn't work and I have no idea why

EDIT: this only seems to be a problem (for me) when I run fastboot-app-server locally... When it runs in prod, no moment errors, and when it runs locally via ember-cli-fastboot, no moment errors. So I'm truly at a loss.

**Comment by michelvermeer** [comment]
It seems that fastboot does not handle transformations accordingly.
To fix this, add the following to your ember-cli-build.js

`app.import('node_modules/moment/moment.js', {
		using: [
			{ transformation: 'amd', as: 'moment' }
		]
	});`

**Comment by billdami** [comment]
I have this same issue running ember-cli-fastboot 2.1.1 and fastboot-app-server 1.1.0, both in dev _and_ production environments. It does not occur when just running the fastboot dev server (via `ember serve`) locally.

@michelvermeer's fix above to manually import moment.js in ember-cli-build.js resolves the issue for me.

**Comment by antonbavykin1991** [comment]
maybe just need to update version of release, 
`https://github.com/jasonmit/ember-cli-moment-shim/blob/master/addon/lib.js`
vs
`https://github.com/jasonmit/ember-cli-moment-shim/blob/v3.7.1/addon/lib.js`

try to check with my fork, in your package.json
`"ember-cli-moment-shim": "git://github.com/antonbavykin1991/ember-cli-moment-shim.git"`

**Comment by jasonmit** [comment]
This should be resolved now in 3.8.0
